src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp

```diff
@@ -2112,61 +2113,95 @@ QgsProcessingExpressionParameterDefinitionWidget::QgsProcessingExpressionParamet
   vlayout->setContentsMargins( 0, 0, 0, 0 );
   vlayout->addWidget( new QLabel( tr( "Default value" ) ) );
 
-  mDefaultLineEdit = new QgsExpressionLineEdit();
-  mDefaultLineEdit->registerExpressionContextGenerator( this );
-
+  mDefaultLineEdit = new QLineEdit();
```

Comment:
If we want to keep exposing the proper widget here, we could use a stacked widget with both a QgsExpressionLineEdit and the point cloud equivalent as different pages. Then the page would be changed when the mExpressionTypeComboBox selection changes.

Reply:
Thanks for the suggestion, implemented it this way.